lib/ai_bot/personas/persona.rb

```diff
@@ -171,6 +176,68 @@ def find_tool(parsed_function)
             persona_options: options[tool_klass].to_h,
           )
         end
+
+        def rag_fragments_prompt(conversation_context)
+          upload_refs =
+            UploadReference.where(target_id: id, target_type: "AiPersona").pluck(:upload_id)
+
+          return nil if !SiteSetting.ai_embeddings_enabled?
+          return nil if conversation_context.blank? || upload_refs.blank?
+
+          latest_interactions =
+            conversation_context
+              .select { |ctx| %i[model user].include?(ctx[:type]) }
+              .map { |ctx| ctx[:content] }
+              .last(10)
+              .join("\n")
+
+          strategy = DiscourseAi::Embeddings::Strategies::Truncation.new
+          vector_rep =
+            DiscourseAi::Embeddings::VectorRepresentations::Base.current_representation(strategy)
+
+          interactions_vector = vector_rep.vector_from(latest_interactions)
+
+          candidate_fragment_ids =
+            vector_rep.asymmetric_rag_fragment_similarity_search(
+              interactions_vector,
+              persona_id: id,
+              limit: 50,
+              offset: 0,
+            )
+
+          guidance =
+            RagDocumentFragment.where(upload_id: upload_refs, id: candidate_fragment_ids).pluck(
+              :fragment,
+            )
+
+          if DiscourseAi::Inference::HuggingFaceTextEmbeddings.reranker_configured?
```

Comment:
Do we want to test for this a few lines before and only get 10 lines when this isn't available?